python/ray/data/datasource/tfrecords_datasource.py

```diff
@@ -7,30 +8,93 @@
     Union,
     Iterable,
     Iterator,
+    List,
 )
 import struct
 
-import numpy as np
-
 from ray.util.annotations import PublicAPI
 from ray.data._internal.util import _check_import
+from ray.data.datastream import Datastream
```

Comment:
@scottjlee wondering why this leads to `ImportError: cannot import name 'Datastream' from 'ray.data.datastream' (/ray/python/ray/data/datastream.py)`